### DISCUSSION TITLE
Dialogs not opening

### DISCUSSION THREAD

**Discussion by s8foschm** [VSCodium/vscodium#1999] [Q&A]
I have had an issue for the last few days, where vscodium on my system does not open any dialogs. When I want to open a file, trying 3 different options (file menu -> open file, Ctrl + O, and entering "open file" in the command palette), nothing happens. Same goes for the "Save as" menu.
What could cause this? Also, is there a way I could find error logs or something like that, so that I can provide more information?
I am running the latest version of VSCode from the AUR, on a Manjaro system with i3 Window Manager. Everything is up to date.

**Comment by Jeong-ilKajokaya** [comment]
Hi! It sounds like you are running into an issue related to **dialog handling in VSCodium/VSCode on i3 WM**. This is a known problem sometimes caused by the combination of Wayland/X11 settings, GTK versions, or the window manager not allowing modal dialogs to open correctly.

Here’s how you can debug and potentially fix it:

---

### 1. Check the logs

You can start VSCodium from a terminal to see error messages:

```bash
codium --verbose
```

or check the main log file if it exists:

```bash
~/.config/VSCodium/logs/
```

Look for entries related to dialogs or GTK errors.

---

### 2. Test with a clean profile

Sometimes extensions or settings can interfere. Try running:

```bash
codium --user-data-dir /tmp/vscodium-test
```

Then try opening a file or saving as. If it works, the problem is likely in your config or extensions.

---

### 3. Check window manager / GTK issues

* i3 sometimes blocks modal dialogs depending on how it handles transient windows.
* Try running VSCodium with XWayland explicitly if you’re on Wayland:

```bash
env GDK_BACKEND=x11 codium
```

* Make sure your GTK packages are up to date:

```bash
sudo pacman -Syu gtk3 gtk4
```

---

### 4. Try alternative dialog handling

* Some users workaround this by launching VSCodium with a specific environment variable:

```bash
export VSCODE_USE_X11=true
codium
```

This forces the app to use X11 dialogs instead of any Wayland/GTK ones that may misbehave.

---

**Summary:**

* Open VSCodium from terminal (`--verbose`) to check errors.
* Test with a clean user-data-dir to rule out extensions/configs.
* Try forcing X11 or updating GTK packages.
* i3 WM sometimes requires extra config for modal dialogs to work correctly.

These steps should help you identify the underlying cause and provide logs for further help.
